### PR TITLE
Replace message scanning with slash commands for network dump uploads

### PR DESCRIPTION
**Pull request by jonbarrow**
I was not aware before that slash commands could be used to upload attachments. Rather than scanning messages, which can be faulty and pose some security concerns, using slash commands give us a bit more control and are safer for the end user

The docs have not been updated yet. Only tested with proxy dump uploads, but it's the same logic for all of them so it should work

### CODE REVIEW

**Comment by hauntii** [comment]
Looks good to me, although it's probably good if we alert the user if they unsafely upload dumps to use the command and delete their message rather than ignoring it.

**Comment by jonbarrow** [comment]
> it's probably good if we alert the user

I already deleted the file :(

File: src/commands/upload-network-dump.js [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+		],
+		message: `<@${interaction.member.id}> Uploaded a WireShark PCAP: ${username}:${password}\n\n${description}`
+	};
+}
+
+/**
+ *
+ * @param {Discord.CommandInteraction} interaction
+ */
+async function hokakuHTTPHandler(interaction) {
```
Comment (DaniElectra) [inline src/commands/upload-network-dump.js:219]: this isn't "Hokaku" technically

Comment (jonbarrow) [reply src/commands/upload-network-dump.js:219]: Copy paste, oops

File: src/commands/upload-network-dump.js [hunk 2]
```diff
@@ -0,0 +1,369 @@
+const path = require('node:path');
+const Discord = require('discord.js');
+const database = require('../database');
+const { SlashCommandBuilder } = require('@discordjs/builders');
+
+const HOKAKU_CAFE_DEFAULT_PCAP_NAME_REGEX = /\d{4}-\d{2}-\d{2}_\d{2}-\d{2}-\d{2}\.(?:pcapng|pcap)/;
+const HOKAKU_CAFE_BIN_NAME_REGEX = /nexServiceToken-\d{10}-[\dA-f]{8}\.bin/;
+const HOKAKU_CTR_DEFAULT_PCAP_NAME_REGEX = /\d{6}_\d{6}\.(?:pcapng|pcap)/;
+const MITMPROXY_NINTENDO_DEFAULT_NAME_REGEX = /(?:wiiu|3ds)-latest\.(?:har|chls|chlsj|chlsx|saz)/;
```
[the remark this answers is not in the document]
Comment (jonbarrow) [reply src/commands/upload-network-dump.js:9]: It started off as just HAR, honestly I don't remember why I added the others later. Oops

**Review by DaniElectra — APPROVED** [review]
looks good

**Review by hauntii — APPROVED** [review]

**Pull request merged**